### PR TITLE
Bring Over State V2 Tests From Altair Branch

### PR DESCRIPTION
**Pull request by rauljordan**
Part of #8638, this PR brings over some tests from the hf1 branch

### CODE REVIEW

File: beacon-chain/state/v2/state_trie_block_box_test.go [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+		error string
+	}
+	initTests := []test{
+		{
+			name:  "nil state",
+			state: nil,
+			error: "received nil state",
+		},
+		{
+			name: "nil validators",
```
Comment (rkapka) [inline beacon-chain/state/v2/state_trie_block_box_test.go:34]: No error in this case?

File: beacon-chain/state/v2/state_trie_block_box_test.go [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
```
Comment (rkapka) [inline beacon-chain/state/v2/state_trie_block_box_test.go:77]: No error in this case?

File: beacon-chain/state/v2/state_trie_block_box_test.go [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+				assert.ErrorContains(t, tt.error, err)
+			} else {
+				assert.NoError(t, err)
+			}
+		})
+	}
+}
+
+func TestBeaconState_HashTreeRoot(t *testing.T) {
+	t.Skip("TODO: Fix FSSZ HTR for sync committee and participation roots")
```
Comment (rkapka) [inline beacon-chain/state/v2/state_trie_block_box_test.go:105]: Any update on this?

File: beacon-chain/state/v2/state_trie_block_box_test.go [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+			if len(oldHTR) != 0 && bytes.Equal(root[:], oldHTR) {
+				t.Errorf("Expected HTR to change, received %#x == old %#x", root, oldHTR)
+			}
+			oldHTR = root[:]
+		})
+	}
+}
+
+func TestBeaconState_HashTreeRoot_FieldTrie(t *testing.T) {
+	t.Skip("TODO: Fix FSSZ HTR for sync committee and participation roots")
```
Comment (rkapka) [inline beacon-chain/state/v2/state_trie_block_box_test.go:174]: Any update on this?

**Comment by rauljordan** [comment]
@prestonvanloon already has a PR for this in #8716 that is blocked by ssz it seems

**Pull request closed without merging**